common/src/main/java/hunternif/mc/impl/atlas/core/scanning/TileDetectorBase.java

```diff
@@ -41,36 +47,54 @@ public class TileDetectorBase implements ITileDetector {
-    /**
-     * Set to true for biome IDs that return true for BiomeDictionary.isBiomeOfType(BEACH)
-     */
-    private static final Set<Identifier> beachBiomes = new HashSet<>();
-
-    private static final Set<Identifier> swampBiomes = new HashSet<>();
-
-    /**
-     * Scan all registered biomes to mark biomes of certain types that will be
-     * given higher priority when identifying mean biome ID for a chunk.
-     * (Currently WATER, BEACH and SWAMP)
-     */
-    public static void scanBiomeTypes() {
-        for (Biome biome : BuiltinRegistries.BIOME) {
-            switch (biome.getCategory()) {
-                case BEACH -> beachBiomes.add(BuiltinRegistries.BIOME.getId(biome));
-                case RIVER, OCEAN -> waterBiomes.add(BuiltinRegistries.BIOME.getId(biome));
-                case SWAMP -> swampBiomes.add(BuiltinRegistries.BIOME.getId(biome));
-            }
-        }
+//  Seems like init-time biome classification isn't a thing anymore, so, can't cache these, at least not out of world.
+//    /**
+//     * Set to true for biome IDs that return true for BiomeDictionary.isBiomeOfType(WATER)
+//     */
+//    private static final Set<Identifier> waterBiomes = new HashSet<>();
+//    /**
+//     * Set to true for biome IDs that return true for BiomeDictionary.isBiomeOfType(BEACH)
+//     */
+//    private static final Set<Identifier> beachBiomes = new HashSet<>();
+//
+//    private static final Set<Identifier> swampBiomes = new HashSet<>();
+//
+//    /**
+//     * Scan all registered biomes to mark biomes of certain types that will be
+//     * given higher priority when identifying mean biome ID for a chunk.
+//     * (Currently WATER, BEACH and SWAMP)
+//     */
+//    public static void scanBiomeTypes() {
+//        for (Biome biome : BuiltinRegistries.BIOME) {
+//            switch (biome.getCategory()) {
+//                case BEACH -> beachBiomes.add(BuiltinRegistries.BIOME.getId(biome));
+//                case RIVER, OCEAN -> waterBiomes.add(BuiltinRegistries.BIOME.getId(biome));
+//                case SWAMP -> swampBiomes.add(BuiltinRegistries.BIOME.getId(biome));
+//            }
+//        }
+//    }
+//
+//    int priorityForBiome(Identifier biome) {
+//        if (waterBiomes.contains(biome)) {
+//            return 4;
+//        } else if (beachBiomes.contains(biome)) {
+//            return 3;
+//        } else {
+//            return 1;
+//        }
+//    }
+
+    @ExpectPlatform
+    static private boolean hasSwampWater(RegistryEntry<Biome> biomeTag) {
+        throw new AssertionError("Not implemented");
     }
 
-    int priorityForBiome(Identifier biome) {
-        if (waterBiomes.contains(biome)) {
+    public static int priorityForBiome(World world, Biome biome) {
+        Registry<Biome> biomeRegistry = world.getRegistryManager().get(Registry.BIOME_KEY);
+        RegistryEntry<Biome> biomeTag = biomeRegistry.entryOf(biomeRegistry.getKey(biome).orElse(null));
```

Comment:
While I appreciate the new biome tags in general, I feel like getting to the `biomeTag` is such a pain in the ass. This makes me question whether this function get's the correct input.

Reply:
I've been playing around with the mod for a while and I'm definitely getting correct biomes on my overworld maps. So weird as it is, I think so? It's the same method used elsewhere in the code it just looks janky here.

Reply:
You misunderstood me here. I was thinking about the method parameters. In particular, if `Biome` is the correct type, or if it should be a BiomeKey or something like that instead.

Reply:
You might be getting mappings mixed up - in yarn it's `Biome` - the closest thing to `BiomeKey` is `BiomeKeys`, which is used to register and define all the Vanilla biomes to the BIOME_KEY registry (similar to BiomeTags or Items)

Reply:
Let me rephrase.  What hits me is the over-and-over repeated code to get a biomeTag from a biome.
In particular, this method gets called from `getBiomeID`. At call time, there already is the biomeTag available, but the biome is passed instead. Leading to this method having a duplication of code and being a bit less efficient because it has to do the lookup again. Having said that, from what the method name suggests, one would not expect to pass in a biomeTag instance, but a little rename would fix that.

Reply:
Yep, got it, that's on me - the tiny diff github was giving me here was leading me astray, swapped to just `RegistryEntry<Biome>` instead of passing in both world and biome in 733485f